flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/utils/LogicalTypeMerging.java

```diff
@@ -301,6 +301,16 @@ public static LogicalType findSumAggType(LogicalType argType) {
      * integral part of a result from being truncated.
      *
      * <p>https://docs.microsoft.com/en-us/sql/t-sql/data-types/precision-scale-and-length-transact-sql
+     *
+     * <p>The rules although inspired by SQL Server they are not followed 100%, instead the approach
```

Comment:
`The rules (although inspired by SQL Server) are not followed 100%`

Reply:
fixed, also added more explanation.